compiler/modules/bir/bir.bal

```diff
@@ -584,30 +584,42 @@ public type EqualityInsn readonly & record {|
     Operand[2] operands;
 |};
 
-# Call a function.
-# This is a not a terminator.
-# This is a PPI. A panic in the called function
+public type FunctionOperand FunctionConstOperand|Register;
+# Call a function. In most cases we don't need to worry about the specific call
+# instruction we have used and instead treat it as this common type.
+# Call instructions are not terminators.
+# Call instructions are PPI. A panic in the called function
 # goes to the onPanic label in the basic block.
 # Regardless of where the function itself panics,
 # any function call could result in a stack overflow panic.
-# XXX This does not handle functions that don't return
+# If the function type is atomic we represent the parameters as a tuple type.
+# In the case of CallConstInsn, arguments corresponding to the rest type of that tuple
+# are expected to be given as a single list value. In the case of CallIndirectInsn,
+# fallowing this convention for arguments is optional and must be explicitly
+# specified by the restParamIsList.
+# XXX These do not handle functions that don't return
 # (i.e. with return type of never)
-public type CallInsn readonly & record {|
+public type CallInsn readonly & record {
```

Comment:
The suggestion was to call this CallInsnBase, so the other can be CallInsn.

Reply:
Fixed